### PR TITLE
ci: add logql-analyzer CI

### PR DESCRIPTION
**Pull request by trevorwhitney**
**What this PR does / why we need it**:

Adds `logql-analyzer` CI that got removed with the drone removal.

**Checklist**
- [ ] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
  - **Note** that Promtail is considered to be feature complete, and future development for logs collection will be in [Grafana Alloy](https://github.com/grafana/alloy). As such, `feat` PRs are unlikely to be accepted unless a case can be made for the feature actually being a bug fix to existing behavior.
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: .github/workflows/logql-analyzer.yml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      run: |
+        echo $("${WORKSPACE_ROOT}/loki/tools/image-tag") > .tag
+        export RELEASE_TAG=$(cat .tag)
+        # if the tag matches the pattern `D.D.D` then RELEASE_NAME="D-D-x", otherwise RELEASE_NAME="next"
+        export RELEASE_NAME=$([[ $RELEASE_TAG =~ $RELEASE_TAG_REGEXP ]] && echo $RELEASE_TAG | grep -oE $MAJOR_MINOR_VERSION_REGEXP | sed "s/\\./-/g" | sed "s/$/-x/" || echo "next")
+        echo $RELEASE_NAME
+        echo $PLUGIN_CONFIG_TEMPLATE > updater-config.yaml
+        # replace placeholders with RELEASE_NAME and RELEASE TAG
+        sed -i "s/\\"{{release}}\\"/\\"$RELEASE_NAME\\"/g" %s % configFileName
+        sed -i "s/{{version}}/$RELEASE_TAG/g" %s % configFileName
```
Comment (vlad-diachenko) [inline .github/workflows/logql-analyzer.yml:69]: we do not need this line because the config is created on the next step

Comment (vlad-diachenko) [reply .github/workflows/logql-analyzer.yml:69]: also, it's a JSONNET ;)

File: .github/workflows/logql-analyzer.yml [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+          "git_committer_name": "version_bumper[bot]",
+          "pull_request_branch_prefix": "logql-analyzer/updater",
+          "pull_request_enabled": true,
+          "pull_request_existing_strategy": "replace",
+          "pull_request_title_prefix": "[logql-analyzer updater] ",
+          "pull_request_message": "Add logql-analyzer version to ${RELEASE_VERSION} to supported versions",
+          "update_jsonnet_attribute_configs": [
+              {
+                "file_path": "ksonnet/environments/logql-analyzer/supported-versions.libsonnet",
+                "jsonnet_key": "${RELEASE_NAME}",
```
Comment (vlad-diachenko) [inline .github/workflows/logql-analyzer.yml:93]: are ENV variables shared between the steps? 
I see that `Run shell command` step exports it but I am not sure if they are available on this step

**Review by vlad-diachenko — APPROVED** [review]
I believe that we need to address the comments before the merge.

File: .github/workflows/logql-analyzer.yml (hunk 2 above)
Comment (trevorwhitney) [reply .github/workflows/logql-analyzer.yml:97]: nope, good call, they either need to be echoed into `$GITHUB_ENV` or set using the `::set-output` syntax, which I opted for

**Pull request merged**